thirdweb/modules/currency.py

```diff
@@ -149,6 +150,18 @@ def burn_from(self, from_address: str, amount: int):
         return self.execute_tx(self.__abi_module.burn_from.build_transaction(
             from_address, amount, self.get_transact_opts()
         ))
+    def transfer(self, to_address: str, amount: int):
+        """ 
+        :param to_address: The address to transfer to
+        :param amount: The amount to transfer
+
+        Transfers the given amount from the current address
+        """
+        if(self.__abi_module.is_restricted_transfer()):
```

Comment:
you don't need to do this manually because when you execute_tx it'd fail early and we don't have to add 1 tx overhead to the call. it's better we try catch on execute_tx and check the error and see if it's restricted then throw that error rather than enforcing the logic here on the client side.
once you remove this we can merge this.